File: Android/MeetingManagerAndroid/app/src/main/java/com/microsoft/office365/meetingmgr/Models/Attendee.java
package com.microsoft.office365.meetingmgr.Models;

/**
 * Representation of Meeting attendee
 */
public class Attendee {
    public EmailAddress EmailAddress = new EmailAddress();
    public ResponseStatus Status;
    public String Type;

    Attendee() {}

    public Attendee(ADUser user) {
        EmailAddress.Address = user.userPrincipalName;
        EmailAddress.Name = user.displayName;
    }

    public Attendee(EmailAddress address) {
        EmailAddress= address;
    }

    @Override
    public String toString() {
        return EmailAddress.toString();
    }
}

File: UWP/Meeting_Manager_UWP/MeetingManager/Models/Attendee.cs
//Copyright (c) Microsoft. All rights reserved. Licensed under the MIT license.
//See LICENSE in the project root for license information.

using Newtonsoft.Json;
using Prism.Windows.AppModel;
using Windows.ApplicationModel.Resources;

namespace MeetingManager.Models
{
    public class Attendee
    {
        public EmailAddress EmailAddress { get; set; }
        public ResponseStatus Status { get; set; }
        public string Type { get; set; }

        [JsonIgnore]
        public string ResponseStatus
        {
            get
            {
                if (IsOrganizer)
                {
                    return ResMan.GetString("OrganizerResponse");
                }
                else
                {
                    if (Status != null)
                    {
                        switch (Status.Response.ToLower())
                        {
                            case OData.Accepted:
                                return ResMan.GetString("AcceptedResponse");
                            case OData.TentativelyAccepted:
                                return ResMan.GetString("TentativeResponse");
                            case OData.Declined:
                                return ResMan.GetString("DeclinedResponse");
                        }
                    }

                    return ResMan.GetString("NoResponse");
                }
            }
        }

        [JsonIgnore]
        public string Name => ToString();

        [JsonIgnore]
        public bool IsOrganizer => EmailAddress.Address.EqualsCaseInsensitive(OrganizerAddress);

        [JsonIgnore]
        public string OrganizerAddress { get; set; }

        public override string ToString()
        {
            return EmailAddress.ToString();
        }
    }
}
